tmt/base.py

```diff
@@ -161,8 +161,8 @@ def from_spec(cls, raw: _RawFmfId) -> 'FmfId':
         # TODO: with mandatory validation, this can go away.
         ref = raw.get('ref', None)
         if not isinstance(ref, (type(None), str)):
-            raise tmt.utils.SpecificationError(
-                f"The 'ref' field must be a string, got '{type(ref).__name__}'.")
+            # TODO: deliver better key address
+            raise tmt.utils.NormalizationError('ref', ref, 'not set or string')
```

Comment:
This would result in a message:
```
Field 'ref' can be not set or string, 'int' found.
```
I think it is gramatically incorrect and ambiguous, it could be interpreted as `it can be not (set or string)` which is unintended.

I'd say `unset` is better than `not set`:
```
Field 'ref' can be unset or string, 'int' found.
```

Reply:
Fixed in ede71004.